// snaptrade-typescript-sdk/dist/client.js
"use strict";
/*
SnapTrade

Connect brokerage accounts to your app for live positions and trading

The version of the OpenAPI document: 1.0.0
Contact: api@snaptrade.com

NOTE: This file is auto generated by Konfig (https://konfigthis.com).
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.Snaptrade = void 0;
const api_1 = require("./api");
const configuration_1 = require("./configuration");
const client_custom_1 = require("./client-custom");
class Snaptrade extends client_custom_1.SnaptradeCustom {
    constructor(configurationParameters = {}) {
        super(configurationParameters);
        const configuration = new configuration_1.Configuration(configurationParameters);
        this.accountInformation = new api_1.AccountInformationApi(configuration);
        this.apiStatus = new api_1.ApiStatusApi(configuration);
        this.authentication = new api_1.AuthenticationApi(configuration);
        this.connections = new api_1.ConnectionsApi(configuration);
        this.options = new api_1.OptionsApi(configuration);
        this.referenceData = new api_1.ReferenceDataApi(configuration);
        this.trading = new api_1.TradingApi(configuration);
        this.transactionsAndReporting = new api_1.TransactionsAndReportingApi(configuration);
    }
}
exports.Snaptrade = Snaptrade;
