// snaptrade-typescript-sdk/dist/configuration.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/*
SnapTrade

Connect brokerage accounts to your app for live positions and trading

The version of the OpenAPI document: 1.0.0
Contact: api@snaptrade.com

NOTE: This file is auto generated by Konfig (https://konfigthis.com).
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.Configuration = void 0;
class Configuration {
    constructor(param = {}) {
        var _a;
        this.consumerKey = param.consumerKey;
        this.apiKey = param.apiKey;
        if (this.apiKey === undefined) {
            this.apiKey = {};
            if (param.clientId !== undefined)
                this.apiKey["clientId"] = param.clientId;
            if (param.signature !== undefined)
                this.apiKey["signature"] = param.signature;
            if (param.timestamp !== undefined)
                this.apiKey["timestamp"] = param.timestamp;
        }
        this.username = param.username;
        this.password = param.password;
        this.accessToken = param.accessToken;
        this.basePath = param.basePath;
        this.baseOptions = (_a = param.baseOptions) !== null && _a !== void 0 ? _a : {};
        this.userAgent = param.userAgent === undefined ? "Konfig/9.0.113/typescript" : param.userAgent;
        this.formDataCtor = param.formDataCtor;
    }
    /**
     * Check if the given MIME is a JSON MIME.
     * JSON MIME examples:
     *   application/json
     *   application/json; charset=UTF8
     *   APPLICATION/JSON
     *   application/vnd.company+json
     * @param mime - MIME (Multipurpose Internet Mail Extensions)
     * @return True if the given MIME is JSON, false otherwise.
     */
    isJsonMime(mime) {
        const jsonMime = new RegExp('^(application\/json|[^;/ \t]+\/[^;/ \t]+[+]json)[ \t]*(;.*)?$', 'i');
        return mime !== null && (jsonMime.test(mime) || mime.toLowerCase() === 'application/json-patch+json');
    }
}
exports.Configuration = Configuration;
